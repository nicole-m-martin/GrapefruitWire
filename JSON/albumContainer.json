{
    "recording-offset": 0,
    "recording-count": 5,
    "recordings": [
    {
    "title": "Red Light Special (radio edit)",
    "length": 284000,
    "disambiguation": "",
    "id": "10648c2d-15f4-4973-bb8f-abd930fc7e7e",
    "first-release-date": "1995-02-27",
    "video": false
    },
    {
    "title": "My Secret Enemy",
    "length": 335000,
    "disambiguation": "",
    "first-release-date": "1995-02-27",
    "id": "2bd3b0b0-b569-49ef-aa28-92c08d79cc77",
    "video": false
    },
    {
    "length": 271000,
    "title": "Red Light Special (L.A.'s Flava mix)",
    "id": "a8ea5e02-7d7e-4c7b-9fff-512a25334240",
    "first-release-date": "1995-02-27",
    "disambiguation": "",
    "video": false
    },
    {
    "video": false,
    "disambiguation": "",
    "first-release-date": "1995-02-27",
    "id": "b2eb86fd-23b7-486e-bf22-c0db8e0cf798",
    "title": "Red Light Special (Gerald Hall's remix)",
    "length": 312000
    },
    {
    "length": 306000,
    "title": "Red Light Special (album version)",
    "disambiguation": "",
    "first-release-date": "1995-02-27",
    "id": "ca491b28-634f-4956-960a-8aad0b5c0538",
    "video": false
    }
    ]
    }