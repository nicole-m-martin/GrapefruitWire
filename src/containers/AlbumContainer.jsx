import React, { useState, useEffect } from 'react';
import { useParams } from 'react-router';
import AlbumSongList from '../components/album/AlbumSongList';
import { fetchSongs } from '../services/brainz-api';

const AlbumContainer = () => {
  const { albumId } = useParams();
  const [loading, setLoading] = useState(true);
  const [songs, setSongs] = useState([]);

  useEffect(() => {
    fetchSongs(albumId)
      .then(setSongs)
      .finally(() => setLoading(false));
  }, []);

  return loading ? (
    <h2>Loading...</h2>
  ) : (
    <main>
      <h1>Tracklist</h1>
      <AlbumSongList songs={songs} />
    </main>
  );
};

export default AlbumContainer;
