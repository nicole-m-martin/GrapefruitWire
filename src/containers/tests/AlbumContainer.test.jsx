import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { MemoryRouter } from 'react-router';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import albumContainer from '../../../JSON/albumContainer.json';
import AlbumContainer from '../AlbumContainer';

const server = setupServer(
  rest.get('http://musicbrainz.org/ws/2/recording?release=027eea17-113a-4355-8c9b-5a12e08b7275&fmt=json', (req, res, ctx) => {
    return res(ctx.json(albumContainer));
  })
);

describe('album container', () => {
  beforeAll(() => server.listen());
  afterAll(() => server.close());

  it('fetches albums adn renders songs to the page', () => {
    render(
      <MemoryRouter>
        <AlbumContainer />
      </MemoryRouter>
    );

    const ul = screen.findByRole('list', { name: 'songs' });

    expect(ul).toMatchSnapshot();
  });
});
