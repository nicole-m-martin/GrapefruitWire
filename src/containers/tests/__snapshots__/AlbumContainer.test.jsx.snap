// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`album container fetches albums adn renders songs to the page 1`] = `Promise {}`;
